core/messaging/__init__.py

```diff
@@ -32,8 +38,35 @@ def __getitem__(self, item):
         return getattr(self, item)
 
     @property
-    def cell_indexes(self) -> str:
-        return ", ".join(self.__cell_indexes)
+    def cell_indexes(self) -> tuple[str, ...] | None:
+        return self.__cell_indexes
+
+    @classmethod
+    def __clean_cell_indexes(cls, cell_indexes) -> tuple[str, ...] | None:
+        """Remove any duplicate cell references, sorts them in column-then-row order, and ensures uppercase columns"""
+        if cell_indexes is None:
+            return None
+        if None in cell_indexes:
+            raise NotImplementedError("`cell_indexes` cannot contain None elements; must be None directly")
```

Comment:
Maybe `ValueError` would be more appropriate?

Reply:
ha yes, also in agreement, fixed 👍